rfcs/2023-05-03-data-volume-metrics.md

```diff
@@ -0,0 +1,146 @@
+# RFC 2023-05-02 - Data Volume Insights metrics
+
+Vector needs to be able to emit accurate metrics that can be usefully queried
+to give users insights into the volume of data moving through the system.
+
+## Scope
+
+### In scope
+
+- All event metrics within Vector need to emit the estimated JSON size of the
+  event. With a consistent method for determining the size it will be easier
+  to accurately compare data in vs data out.
```

Comment:
Not _all_, just the volume ones:

* `component_received_event_bytes_total`
* `component_sent_event_bytes_total`
* `component_received_event_total`
* `component_sent_event_total`